update bitcoin-tx to output witness data

jnewbery opened the pull request:
In https://github.com/bitcoin/bitcoin/commit/7c4bf779e8b74e474551982a24f5acc265293abd , jl2012 updated getrawtransaction() to return the witness data for transaction inputs spending P2WPK and P2WSH.

bitcoin-tx doesn't call into this function and instead calls `TxToUniv()` in `core_write.cpp`, so bitcoin-tx calls don't return witness data.

I think the proper fix is to have `getrawtransaction()` call into `TxToUniv()` so we don't have duplicate code. For now, I'm just adding code to `TxToUniv()` to return witness data.

jnewbery committed:
update bitcoin-tx to output witness data

sdaftuar commented on src/core_write.cpp:
```diff
@@ -166,6 +167,16 @@ void TxToUniv(const CTransaction& tx, const uint256& hashBlock, UniValue& entry)
+                        std::vector<unsigned char> item = tx.wit.vtxinwit[i].scriptWitness.stack[j];
+                        txinwitness.push_back(HexStr(item.begin(), item.end()));
+                    }
+                in.push_back(Pair("txinwitness", txinwitness));
```
style nit: indentation

jnewbery replied:
Fixed (and changed `push_back(Pair())` to `pushKV` to match other fields).

All test_bitcoin tests pass.

sdaftuar reviewed (changes requested):
utACK apart from the one style nit -- I agree with you that it seems like we should use this function in the RPC call as well.  Maybe when you open a PR upstream you should ask if it'd be better to add an additional commit that does the deduplication you mentioned?  (I haven't checked to see if there would be any changes to the output by doing so,  but I can't think of why it would be a problem to de-duplicate the code somehow.)

jnewbery commented:
I've addressed @sdaftuar's code nit and also added the witness hash to the `TxToUniv()` output.

Suhas, @TheBlueMatt , do you have any further comments before I open a PR against https://github.com/bitcoin/bitcoin/ ? I'm planning to open this PR and also https://github.com/HackerResidency/bitcoin/pull/4 to deduplicate the `TxToUniv`/`TxToJSON` code.

TheBlueMatt commented:
utACK.

jnewbery commented:
Thanks. All squashed. I'll open a PR to bitcoin/bitcoin.

jnewbery commented:
I've opened a PR to bitcoin/bitcoin. Closing this PR.
